src/test/java/com/joyent/manta/archiver/ManataClientSupplierTest.java

```diff
@@ -0,0 +1,296 @@
+
+import jline.internal.Log;
+
+@Test
+public class ManataClientSupplierTest {
+    private static Logger LOG = LoggerFactory.getLogger(ManataClientSupplierTest.class);
+
+    private static final MantaClientSupplier MANTA_CLIENT_SUPPLIER = new MantaClientSupplier();
+    // Seems silly, but what if they change?
+    public static final String MANTA_URL = "MANTA_URL";
+    public static final String MANTA_USER = "MANTA_USER";
+    public static final String MANTA_KEY_ID = "MANTA_KEY_ID";
+    public static final String MANTA_KEY_PATH = "MANTA_KEY_PATH";
+
+    public static final String MANTA_URL_PROPERTY = "manta.url";
+    public static final String MANTA_USER_PROPERTY = "manta.user";
+    public static final String MANTA_KEY_ID_PROPERTY = "manta.key_id";
+    public static final String MANTA_KEY_PATH_PROPERTY = "manta.key_path";
+    public static final String MANTA_PASSWORD_PROPERTY = "manta.password";
+
+    public static final String MANTA_KEY_CONTENT = "manta.key_content";
+    public static final String MANTA_PASSWORD = "manta.password";
+    public static final String MANTA_NO_AUTH = "manta.no_auth";
+    public static final String MANTA_NO_NATIVE_SIGS = "manta.disable_native_sigs";
+    public static final String MANTA_VERIFY_UPLOAD = "manta.verify_uploads";
+    public static final String MANTA_TIMEOUT = "manta.timeout";
+    public static final String MANTA_HTTP_RETRIES = "manta.retries";
+    public static final String MANTA_MAX_CONNS = "manta.max_connections";
+    public static final String MANTA_HTTP_BUFFER_SIZE = "manta.http_buffer_size";
+    public static final String MANTA_HTTPS_PROTOCOLS = "https.protocols";
+    public static final String MANTA_HTTPS_CIPHERS = "https.cipherSuites";
+    public static final String MANTA_TCP_SOCKET_TIMEOUT = "manta.tcp_socket_timeout";
+    public static final String MANTA_CONNECTION_REQUEST_TIMEOUT = "manta.connection_request_timeout";
+    public static final String MANTA_EXPECT_CONTINUE_TIMEOUT = "manta.expect_continue_timeout";
+    public static final String MANTA_UPLOAD_BUFFER_SIZE = "manta.upload_buffer_size";
+    public static final String MANTA_SKIP_DIRECTORY_DEPTH = "manta.skip_directory_depth";
+    public static final String MANTA_CLIENT_ENCRYPTION = "manta.client_encryption";
+    public static final String MANTA_CLIENT_ENCRYPTION_KEY_ID = "manta.encryption_key_id";
+    public static final String MANTA_ENCRYPTION_ALGORITHM = "manta.encryption_algorithm";
+    public static final String MANTA_UNENCRYPTED_DOWNLOADS = "manta.permit_unencrypted_downloads";
+    public static final String MANTA_ENCRYPTION_AUTH_MODE = "manta.encryption_auth_mode";
+    public static final String MANTA_ENCRYPTION_KEY_PATH = "manta.encryption_key_path";
+    public static final String MANTA_ENCRYPTION_KEY_BYTES_64 = "manta.encryption_key_bytes_base";
+    private static final String base_dir = String.format("/%s", System.getenv(MANTA_USER));
```

Comment:
This fails to account for subusers, I think you're looking for `com.joyent.manta.client.MantaUtils#parseAccount`